src/wallet/rpcwallet.cpp

```diff
@@ -3707,9 +3709,18 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
         }
 
         // Check that the user specified fee is sane.
-        if (nFee > nTotalOut) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Fee %s is greater than the sum of outputs %s", FormatMoney(nFee), FormatMoney(nTotalOut)));
-        }
+        // This allows amount=0 (and all amount < nDefaultFee) transactions to use the default network fee
+        // or anything less than nDefaultFee instead of being forced to use a custom fee and leak metadata
+        if (nTotalOut < nDefaultFee) {
+            if (nFee > nDefaultFee) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Small transaction amount %s has fee %s that is greater than the default fee %s", FormatMoney(nTotalOut), FormatMoney(nFee), FormatMoney(nDefaultFee)));
```

Comment:
This is better than what was before, but why should this be an error? Couldn't it be that someone wants to send a memo quickly so pays a large fee?

Reply:
@arielgabizon I implemented what @daira requested. It's the remnant of "absurd/sane" fee checks and prevents people from, for example, accidentally paying 1 ZEC in fees to send 0.00001 ZEC, which seems like a good idea.

Reply:
I'll ack after my comment below is addressed.